src/imagery/i.eodag/i.eodag.py

```diff
@@ -300,82 +396,199 @@ def no_fallback_search(search_parameters, provider):
     search_result = dag.search_iter_page(**search_parameters)
 
     # TODO: Would it be useful if user could iterate through
-    # the pages manually, and look for the product themselves?
+    #       the pages manually, and look for the product themselves?
     try:
         # Merging the pages into one list with all products
-        return [j for i in search_result for j in i]
+        return SearchResult([j for i in search_result for j in i])
     except Exception as e:
-        gs.verbose(e)
+        gs.debug(e)
         gs.fatal(_("Server error, please try again."))
 
 
-def create_products_dataframe(eo_products):
-    result_dict = {"id": [], "time": [], "cloudCover": [], "productType": []}
-    for product in eo_products:
-        for key in result_dict:
-            if key == "time":
-                if (
-                    "startTimeFromAscendingNode" in product.properties
-                    and product.properties["startTimeFromAscendingNode"] is not None
-                ):
+def list_products(products):
+    """Lists products on the Standard Output (Console) stream.
+
+    :param products: EO poducts to be listed
+    :type products: class:'eodag.api.search_result.SearchResult'
+    """
+    columns = ["id", "startTimeFromAscendingNode", "cloudCover", "productType"]
```

Comment:
Maybe we can have output to be in JSON format by default. We can then have two ways of printing, maybe 'l' flag and 'L' flag, the former to just print the main product attributes, and the latter to print all the metadata of a product?

I don't think it would be practical to add a `column` option, because the attributes varies with different products.